Gordon360/Controllers/RecIM/ActivitiesController.cs

```diff
@@ -20,34 +21,22 @@ public ActivitiesController(IActivityService activityService)
             _activityService = activityService;
         }
 
-        ///<summary>Gets a list of all Activities</summary>
+        ///<summary>Gets a list of all Activities by parameter </summary>
         ///<param name="active"> Optional active parameter </param>
+        ///<param name="time"> Optional time parameter </param>
         /// <returns>
         /// All Existing Activities 
         /// </returns>
         [HttpGet]
         [Route("")]
-        public ActionResult<IEnumerable<Activity>> GetActivities(Boolean active = false)
-        {
-            if (active)
-            {
-                var result = _activityService.GetActivitiesByTime(null);
-                return Ok(result);
-            } else
+        public ActionResult<IEnumerable<Activity>> GetActivities([FromQuery] DateTime? time, bool active)
```

Comment:
Not sure exactly what your DB models look like, but `beginTime` or `beginDateTime` might be a clearer parameter in this case since time is a bit generic.

Reply:
In all honesty, there needs to be more re-work on the logic of the time based search of Activities. Mainly because, the registration of the Activity might be closed, but the Activity might still be on-going. #724 mentions a possible implementation where we have a **future** and **past** clause where **future** would check if registration hasn't started and **past** would either just check if the league is completed or if registration has closed. We'll iron out that name when it becomes more official what types of use cases we have.

Reply:
Great, thanks for thinking about it. It doesn't need to be approved by me yet! ;)